Example/ViewController.swift

```diff
@@ -139,6 +137,40 @@ class ViewController: UIViewController, GalleryItemsDataSource, GalleryDisplaced
     }
 }
 
+extension ViewController: GalleryDisplacedViewsDataSource {
+
+    func provideDisplacementItem(atIndex index: Int) -> DisplaceableView? {
+
+        return index < imageViews.count ? imageViews[index] : nil
+    }
+}
+
+extension ViewController: GalleryItemsDataSource {
+
+    func itemCount() -> Int {
+
+        return imageViews.count
+    }
+
+    func provideGalleryItem(_ index: Int) -> GalleryItem {
+
+        let imageView = imageViews[index]
+        return items[imageView]!
+    }
+}
+
+extension ViewController: GalleryItemsDelegate {
+
+    func removeGalleryItem(at index: Int) {
+
+        print("remove item at \(index)")
+
+        let imageView = imageViews[index]
+        imageView.removeFromSuperview()
+        imageViews.remove(at: index)
```

Comment:
Correct, we want to keep the example as simple as possible. In this case, for the sake of "completeness", I'd say let's just remove the item from `items` too. Just to show that the consumer is responsible for managing both of these fields... maybe expanding a bit on the idea with a comment?